src/com/github/Kraken3/AFKPGC/LagScanner.java

```diff
@@ -147,20 +146,11 @@ public void run() {
 					} else {
 						stats.put(enttype.name(), entcost);
 					}
-					if (totalCost >= lagSourceThreshold) {
-						AFKPGC.debug("Chunk ", chunkId, " passed source threshold with tiles and entities");
+					if (totalCost >= lagSourceThreshold && !fullScan) {
 						break; // if we cross the threshold, don't keep adding.
 					}
 				}
 			}
-			StringBuffer sb = new StringBuffer();

```

Comment:
I'm thinking, remove the logging line from the loop that calls testChunk, and instead add a line at 156 that displays the same information + the stats summation in one line.

Then, add an ```else``` at 157 that displays the cached result of the test.